test/php/test_php_fallback.sh

```diff
@@ -0,0 +1,170 @@
+            find_version_from_git_tags "${variable_name}" "${repository}" "${prefix}" "${separator}" "${last_part_optional}"
+        else
+            ((breakfix=breakfix-1))
+            if [ "${breakfix}" = "0" ] && [ "${last_part_optional}" = "true" ]; then
+                declare -g ${variable_name}="${major}.${minor}"
+            else 
+                declare -g ${variable_name}="${major}.${minor}.${breakfix}"
+            fi
+        fi
+    set -e
+}
+
+install_previous_version() {
+    echo -e "\nInstalling Previous Version..."
+    PHP_VERSION=$1
+    find_prev_version_from_git_tags PHP_VERSION https://github.com/php/php-src "tags/php-"
+    PHP_INSTALL_DIR="${PHP_DIR}/${PHP_VERSION}"
+    if [ -d "${PHP_INSTALL_DIR}" ]; then
+        echo "(!) PHP version ${PHP_VERSION} already exists."
+        exit 1
+    fi
+
+    if ! cat /etc/group | grep -e "^php:" > /dev/null 2>&1; then
+        groupadd -r php
+    fi
+    usermod -a -G php "${USERNAME}"
+
+    PHP_URL="https://www.php.net/distributions/php-${PHP_VERSION}.tar.gz"
+
+    PHP_INI_DIR="${PHP_INSTALL_DIR}/ini"
+    CONF_DIR="${PHP_INI_DIR}/conf.d"
+    mkdir -p "${CONF_DIR}";
+
+    PHP_EXT_DIR="${PHP_INSTALL_DIR}/extensions"
+    mkdir -p "${PHP_EXT_DIR}"
+
+    PHP_SRC_DIR="/usr/src/php"
+    mkdir -p $PHP_SRC_DIR
+    cd $PHP_SRC_DIR
+    wget -O php.tar.xz "$PHP_URL"
+    ln -sf /usr/local/php/8.1.27 /usr/local/php/current
+    echo 'export PATH=/usr/local/php/current/bin:$PATH' >> ~/.bashrc
+    . ~/.bashrc
+}
+
+install_php() {  
+    apt-get remove –purge php*
+    apt-get purge php*
+    apt-get autoremove
+    apt-get autoclean
+    apt-get remove dbconfig-php
+    apt-get dist-upgrade
+    # whereis php | while IFS= read -r line; do
+    #     # Remove leading and trailing whitespace from the line
+    #     line=$(echo "$line" | awk '{$1=$1};1')
+    #     # Check if the line is not empty
+    #     if [ -n "$line" ]; then
+    #         # Delete the file or directory
+    #         echo "Deleting: $line"
+    #         rm -rf "$line"
+    #     fi
+    # done
+    rm /usr/local/php/current
+
+    # trying to install a non existent fake binary for a possibly new tag which doesn't have a release yet
+    PHP_VERSION="v1.2.xyz"
+    PHP_INSTALL_DIR="${PHP_DIR}/${PHP_VERSION}"
+    if [ -d "${PHP_INSTALL_DIR}" ]; then
+        echo "(!) PHP version ${PHP_VERSION} already exists."
+        exit 1
+    fi
+
+    if ! cat /etc/group | grep -e "^php:" > /dev/null 2>&1; then
+        groupadd -r php
+    fi
+    usermod -a -G php "${USERNAME}"
+
+    PHP_URL="https://www.php.net/distributions/php-${PHP_VERSION}.tar.gz"
+
+    PHP_INI_DIR="${PHP_INSTALL_DIR}/ini"
+    CONF_DIR="${PHP_INI_DIR}/conf.d"
+    mkdir -p "${CONF_DIR}";
+
+    PHP_EXT_DIR="${PHP_INSTALL_DIR}/extensions"
+    mkdir -p "${PHP_EXT_DIR}"
+
+    PHP_SRC_DIR="/usr/src/php"
+    mkdir -p $PHP_SRC_DIR
+    cd $PHP_SRC_DIR
+
+    wget -O php.tar.xz "$PHP_URL" || install_previous_version "8.2"
```

Comment:
Why do we pass an argument to install_previous_version? This doesn't seem exactly the same like the function in `install.sh`?

Can we make sure the `install_previous_version` are copies are each other?

Reply:
Also, shouldn't the VERSION be `latest` in this case, why 8.2?

Reply:
corrected !